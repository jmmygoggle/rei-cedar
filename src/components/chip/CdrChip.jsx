import style from './styles/CdrChip.scss';

export default {
  name: 'CdrChip',
  data() {
    return {
      style,
      baseClass: 'cdr-chip',
    };
  },
  render() {
    return (
      <button
        class={this.style[this.baseClass]}
        {...{ on: this.$listeners }}
      >
        { this.$slots['icon-left'] && (
          <span class={this.style['cdr-chip__icon-left']}>
            { this.$slots['icon-left'] }
          </span>
        )}
        { this.$slots['icon-right'] && (
          <span class={this.style['cdr-chip__icon-right']}>
            { this.$slots['icon-right'] }
          </span>
        )}
        <span class={this.style['cdr-chip__content']}>
          { this.$slots.default }
        </span>
      </button>
    );
  },
};
