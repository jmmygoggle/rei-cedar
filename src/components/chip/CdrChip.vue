<template>
  <button
    :class="$style[this.baseClass]"
  >
    <span v-if="hasIconLeft" :class="$style['cdr-chip__icon-left']">
      <slot name="icon-left" />
    </span>
    <span v-if="hasIconRight" :class="$style['cdr-chip__icon-right']">
      <slot name="icon-right" />
    </span>
    <span :class="$style['cdr-chip__content']">
      <slot />
    </span>
  </button>
</template>

<script lang="ts">
import { defineComponent, computed } from 'vue';
import propValidator from '../../utils/propValidator';
import { buildClass } from '../../utils/buildClass';

export default defineComponent({
  name: 'CdrChip',
  props: {
    type: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['info', 'warning', 'success', 'error'],
      ),
      default: 'info',
    },
  },
  setup(props, ctx) {
    const baseClass = 'cdr-chip';
    const hasIconLeft = ctx.slots['icon-left'];
    const hasIconRight = ctx.slots['icon-right'];
    return {
      baseClass,
      hasIconLeft,
      hasIconRight,
    };
  },
});
</script>

<style lang="scss" module src="./styles/CdrChip.scss">
</style>
