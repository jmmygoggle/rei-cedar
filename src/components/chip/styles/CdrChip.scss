@import '../../../css/settings/index.scss';
@import './vars/CdrChip.vars.scss';

.cdr-chip {
  @include cdr-chip-base-mixin;
}

.cdr-chip__icon-left {
  @include cdr-chip-icon-left-mixin;
}

.cdr-chip__icon-right {
  @include cdr-chip-icon-right-mixin;
}
