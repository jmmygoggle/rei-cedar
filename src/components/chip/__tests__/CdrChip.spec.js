import { shallowMount } from '@vue/test-utils';
import CdrChip from 'componentdir/chip/CdrChip';

describe('CdrChip', () => {
  it('default matches snapshot', () => {
    const wrapper = shallowMount(CdrChip, {
      slots: {
        default: 'tortilla chip'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders icon slots', () => {
    const wrapper = shallowMount(CdrChip, {
      slots: {
        default: 'tortilla chip',
        'icon-left': '🌮',
        'icon-right': '🍔'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
