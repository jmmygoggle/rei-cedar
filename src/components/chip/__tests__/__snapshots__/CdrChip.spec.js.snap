// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrChip default matches snapshot 1`] = `
<button
  class="cdr-chip"
>
  <span
    class="cdr-chip__content"
  >
    tortilla chip
  </span>
</button>
`;

exports[`CdrChip renders icon slots 1`] = `
<button
  class="cdr-chip"
>
  <span
    class="cdr-chip__icon-left"
  >
    🌮
  </span>
  <span
    class="cdr-chip__icon-right"
  >
    🍔
  </span>
  <span
    class="cdr-chip__content"
  >
    tortilla chip
  </span>
</button>
`;
